cpp/src/community/louvain.cu

```diff
@@ -15,25 +15,62 @@
  */
 
 #include <community/louvain.cuh>
+#include <experimental/louvain.cuh>
 
 namespace cugraph {
 
+namespace detail {
+
 template <typename vertex_t, typename edge_t, typename weight_t>
 std::pair<size_t, weight_t> louvain(raft::handle_t const &handle,
-                                    GraphCSRView<vertex_t, edge_t, weight_t> const &graph,
+                                    GraphCSRView<vertex_t, edge_t, weight_t> const &graph_view,
                                     vertex_t *clustering,
                                     size_t max_level,
                                     weight_t resolution)
 {
-  CUGRAPH_EXPECTS(graph.edge_data != nullptr,
+  CUGRAPH_EXPECTS(graph_view.edge_data != nullptr,
                   "Invalid input argument: louvain expects a weighted graph");
   CUGRAPH_EXPECTS(clustering != nullptr, "Invalid input argument: clustering is null");
 
-  Louvain<GraphCSRView<vertex_t, edge_t, weight_t>> runner(handle, graph);
+  Louvain<GraphCSRView<vertex_t, edge_t, weight_t>> runner(handle, graph_view);
+  return runner(clustering, max_level, resolution);
+}
 
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          bool store_transposed,
+          bool multi_gpu>
+std::pair<size_t, weight_t> louvain(
+  raft::handle_t const &handle,
+  experimental::graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu> const
+    &graph_view,
+  vertex_t *clustering,
+  size_t max_level,
+  weight_t resolution)
+{
+  CUGRAPH_EXPECTS(clustering != nullptr, "Invalid input argument: clustering is null");
+
+  experimental::Louvain<
+    experimental::graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu>>
+    runner(handle, graph_view);
   return runner(clustering, max_level, resolution);
 }
 
+}  // namespace detail
+
+template <typename graph_t>
```

Comment:
I guess this graph_t can be either old GraphCSRView or new graph_view_t? Is this a temporary solution to easily switch between the old one and the new one, or something that will stay? Or is this a mechanism to use the old one for SG and the new one for MG? (till we switch to the new class for SG as well)

Reply:
I intend this as an interim step, although it might be a reasonable mechanism to address the templated variations of the new graph_view_t.  Hadn't really thought about that in too much depth, just needed a means for handling the old and new so far.